include/oneapi/mkl/dft/detail/commit_impl.hpp

```diff
@@ -28,35 +28,37 @@
 #endif
 
 #include "oneapi/mkl/detail/backends.hpp"
+#include "types_impl.hpp"
 
 namespace oneapi {
 namespace mkl {
 namespace dft {
 namespace detail {
 
+template <precision prec, domain dom>
 class commit_impl {
 public:
-    commit_impl(sycl::queue queue, mkl::backend backend)
-            : queue_(queue),
-              backend_(backend),
-              status(false) {}
+    commit_impl(sycl::queue queue, mkl::backend backend) : queue_(queue), backend_(backend) {}
 
-    commit_impl(const commit_impl& other) = default;
+    // rule of three
+    commit_impl(const commit_impl& other) = delete;
+    commit_impl& operator=(const commit_impl& other) = delete;
 
     virtual ~commit_impl() = default;
 
     sycl::queue& get_queue() {
         return queue_;
     }
 
-    mkl::backend& get_backend() {
+    mkl::backend get_backend() {
```

Comment:
That makes sense, shouldn't this change move to the main implementation MR? The method could be make `const` if we do this.

Reply:
I think we should make as many methods `const` as possible, in particular in the `include` folder. For instance `descriptor::get_value`. IMO it makes sense to make the change in the main MR, what do you think?

Reply:
done